fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/scheduler/SmsMessageScheduledJobServiceImpl.java

```diff
@@ -285,9 +285,9 @@ public void getDeliveryReports() {
 
                         if (!smsMessageDeliveryReportData.getHasError()
                                 && (deliveryStatus != 100)) {
-                            SmsMessage smsMessage = this.smsMessageRepository.findOne(smsMessageDeliveryReportData.getId());
-                            Integer statusType = smsMessage.getStatusType();
-                            boolean statusChanged = false;
+                            SmsMessage smsMessage = this.smsMessageRepository.findById(smsMessageDeliveryReportData.getId()).orElse(null);
+                            Integer statusType;
+                            boolean statusChanged;
```

Comment:
Why remove this? I'm proposing to put it back, shout if I misunderstood..

```suggestion
                            boolean statusChanged = false;
```

Reply:
I have moved declaration to assignment in the latest commit.